pyrival/algebra/discrete_log.py

```diff
@@ -1,22 +1,50 @@
+from math import gcd, sqrt
+
+
+def extended_gcd(a, b):
+    if not b:
+        return a, 1, 0
+
+    g, y, x = extended_gcd(b, a % b)
+    return g, x, y - ((a // b) * x)
+
+
 def discrete_log(a, b, m):
-    """returns some x s.t. pow(a, x, m) == b or None if no such x exists, works when a amd m are coprime"""
-    a %= m
-    b %= m
-    if b == 1:
+    """returns smallest non-negative x s.t. pow(a, x, m) == b or None if no such x exists"""
+    if b >= m:
+        return None
+
+    if b == 1 or m == 1:
         return 0
```

Comment:
What if a == 0?

Reply:
Hmm what about it? Do you have a specific case in mind that is not working as expected? The method should be able to handle when `a=0` and the tests should cover that too ideally.

Checked the implementation and some cases manually, looks all right to me. For example:
`discrete_log(0, 1, 5) = 0`
`discrete_log(0, 0, 5) = 1`

Reply:
Hm, this is reasonable. Though taking 0^0 = 1 could be a bit of a stretch on some problems.